GeneralsMD/Code/GameEngine/Source/GameLogic/AI/AIPathfind.cpp

```diff
@@ -6159,9 +6159,9 @@ Int Pathfinder::examineNeighboringCells(PathfindCell *parentCell, PathfindCell *
 		ICoord2D newCellCoord;
 		PathfindCell *newCell;
 		const Int adjacent[5] = {0, 1, 2, 3, 0};
-		Bool neighborFlags[8] = {false, false, false, false, false, false, false};
+		Bool neighborFlags[8] = { 0 };
 
-		UnsignedInt newCostSoFar;
+		UnsignedInt newCostSoFar = 0;
```

Comment:
This is not necessary. None of the newCostSoFar are initialized.

Reply:
On the other hand, perhaps it is fine to initialize this now, to avoid this mess in the future. Are there more places with this variable? I thought there are 4 functions with a newCostSoFar, but I may be mistaken.